File: src/2225. Find Players With Zero or One Losses.js
/**
 * @param {number[][]} matches
 * @return {number[][]}
 */

export function findWinners(matches) {
  const lossTimes = new Map()
  for (let [winner, loser] of matches) {
    lossTimes.set(winner, lossTimes.get(winner) ?? 0)
    lossTimes.set(loser, (lossTimes.get(loser) ?? 0) + 1)
  }
  const results = [[], []]
  for (let [player, lossTime] of lossTimes) {
    if (lossTime === 0) results[0].push(player)
    else if (lossTime === 1) results[1].push(player)
  }
  results[0].sort((a, b) => a - b)
  results[1].sort((a, b) => a - b)
  return results
}

File: src/2225. Find Players With Zero or One Losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        int lossesCount[100001];
        fill_n(lossesCount, 100001, -1);
        for (auto& match : matches) {
            int winner = match[0],
                loser = match[1];
            if (lossesCount[winner] == -1) {
                lossesCount[winner] = 0;
            }
            if (lossesCount[loser] == -1) {
                lossesCount[loser] = 1;
            } else {
                lossesCount[loser]++;
            }
        }
        vector<vector<int>> ans(2, vector<int>());
        for (int i = 0; i < 100001; ++i) {
            if (lossesCount[i] == 0) {
                ans[0].push_back(i);
            } else if (lossesCount[i] == 1) {
                ans[1].push_back(i);
            }
        }
        return ans;
    }
};